nemo/collections/asr/models/hybrid_asr_tts_models.py

```diff
@@ -0,0 +1,459 @@
+                asr_model=None,
+                tts_model_path=f"{tts_model_path}",
+                tts_model=None,
+                enhancer_model_path=None,
+                enhancer_model=None,
+                asr_model_type=f"{model_type}",
+                asr_model_fuse_bn=False,  # for training from scratch always should be False
+                train_ds=None,
+                validation_ds=None,
+                test_ds=None,
+                optim=None,
+            )
+        )
+
+        asr_cfg = copy.deepcopy(asr_cfg)  # copy not to avoid original config
+        OmegaConf.set_struct(asr_cfg, False)
+        for subconfig_path in ["optim", "train_ds", "validation_ds", "test_ds"]:
+            if subconfig_path in asr_cfg:
+                cfg[subconfig_path] = asr_cfg.pop(subconfig_path)
+        cfg.asr_model = asr_cfg
+        return cls(cfg=cfg, trainer=trainer)
+
+    @classmethod
+    def from_pretrained_models(
+        cls,
+        asr_model_path: Union[str, Path],
+        tts_model_path: Union[str, Path],
+        asr_model_fuse_bn: bool = False,
+        cfg: Optional[DictConfig] = None,
+        trainer: Optional[Trainer] = None,
+    ):
+        if cfg is None:
+            cfg = DictConfig(
+                dict(
+                    asr_model_path=f"{asr_model_path}",
+                    asr_model=None,
+                    tts_model_path=f"{tts_model_path}",
+                    tts_model=None,
+                    enhancer_model_path=None,
+                    enhancer_model=None,
+                    asr_model_type=None,
+                    asr_model_fuse_bn=asr_model_fuse_bn,
+                    train_ds=None,
+                    validation_ds=None,
+                    test_ds=None,
+                    optim=None,
+                )
+            )
+        else:
+            cfg.tts_model_path = f"{tts_model_path}"
+            cfg.asr_model_path = f"{asr_model_path}"
+        return ASRWithTTSModel(cfg, trainer=trainer)
+
+    def __setattr__(self, name, value):
+        # pytorch-lightning magic, allows to call *_step on asr_model
+        if name == "_current_fx_name" and self._full_init_guard:
+            self.asr_model._current_fx_name = value  # need to make logging inside asr_model work
+        return super().__setattr__(name, value)
+
+    def setup_optimization(
+        self, optim_config: Optional[Union[DictConfig, Dict]] = None, optim_kwargs: Optional[Dict[str, Any]] = None,
+    ):
+        self.tts_model.freeze()
+        optimizer, scheduler = super().setup_optimization(optim_config=optim_config, optim_kwargs=optim_kwargs)
+        # set ASR model optimizer/scheduler to allow training_step on asr_model
+        self.asr_model._optimizer = optimizer
+        self.asr_model._scheduler = scheduler
+        return optimizer, scheduler
+
+    def setup_validation_data(self, val_data_config: Union[DictConfig, Dict]):
+        return self.asr_model.setup_validation_data(val_data_config)
```

Comment:
Can we come up with some kind of alias to save ourselves from manually overriding so many methods?

Reply:
Hm, I have no idea how to do it. The only way I see - is to do it in `__getattr__` method, but this will be ugly. Since I need to implement ASRModel interface, I have do define all these methods to make everything work